src/apps/backend/modules/task/rest-api/task-router.ts

```diff
@@ -13,6 +13,7 @@ export default class TaskRouter extends ApplicationRouter {
     router.post('/', ctrl.createTask);
     router.get('/', ctrl.getTasks);
     router.get('/:id', ctrl.getTask);
+    router.post('/share', ctrl.shareTasks);
```

Comment:
This is not a CRUD API. You are exposing functional API. `share` is an action, not a verb.

You need to model your operations in terms of entities to leverage OOPS, CRUD etc.

`TaskPermission` is one way (corresponding API would be `POST /accounts/:accountId/tasks/:taskId/permissions` and you pass user id and permission. There are other ways too